reference-ril/reference-ril.c

```diff
@@ -627,17 +616,54 @@ static void requestOrSendDataCallList(RIL_Token *t)
     }
 
     at_response_free(p_response);
+    p_response = NULL;
 
-    err = at_send_command_multiline ("AT+CGCONTRDP", "+CGCONTRDP:", &p_response);
-    if (err != 0 || p_response->success == 0) {
-        if (t != NULL)
-            RIL_onRequestComplete(*t, RIL_E_GENERIC_FAILURE, NULL, 0);
-        else
-            RIL_onUnsolicitedResponse(RIL_UNSOL_DATA_CALL_LIST_CHANGED,
-                                      NULL, 0);
-        return;
+    err = at_send_command_multiline ("AT+CGDCONT?", "+CGDCONT:", &p_response);
+    if (err != 0 || p_response->success == 0)
+        goto error;
+
+    for (p_cur = p_response->p_intermediates; p_cur != NULL;
+         p_cur = p_cur->p_next) {
+        char *line = p_cur->line;
+        int cid;
+
+        err = at_tok_start(&line);
+        if (err < 0)
+            goto error;
+
+        err = at_tok_nextint(&line, &cid);
+        if (err < 0)
+            goto error;
+
+        for (i = 0; i < n; i++) {
+            if (responses[i].cid == cid)
+                break;
+        }
+
+        if (i >= n) {
+            /* details for a context we didn't hear about in the last request */
+            continue;
+        }
+
+        // type
+        err = at_tok_nextstr(&line, &out);
+        if (err < 0)
+            goto error;
+
+        if (!strcmp(out, "IP")) {
+            responses[i].type = "IP";
+        } else {
+            responses[i].type = "IPV6";
```

Comment:
Since we only support IP and IPV6 for now, should we just ignore other type? 

maybe:
```
} else if (!strcmp(out, "IPV6")) {
     responses[i].type = "IPV6";
}
```